addons/mail/static/src/thread/message.xml

```diff
@@ -47,12 +47,14 @@
                         <t t-out="message.body"/>
                     </div>
                 </t>
-                <div class="o-mail-message-actions invisible">
+                <div t-if="!props.hideActions" class="o-mail-message-actions invisible">
                     <div class="rounded-1 shadow-sm">
                         <i class="cursor-pointer px-1 py-2 fa fa-lg"
-                            t-att-class="message.isStarred ? 'fa-star' : 'fa-star-o'" 
+                            t-att-class="message.isStarred ? 'fa-star' : 'fa-star-o'"
                             t-on-click="toggleStar"
                             role="button" tabindex="0" title="Mark as Todo" aria-label="Mark as Todo"></i>
+                        <i class="cursor-pointer px-1 py-2 fa fa-lg fa-trash" t-if="canBeDeleted" t-on-click="props.onClickDelete"
```

Comment:
in this case, I think it would be better to have the message itself opening up the confirm dialog, then calling the messaging service to delete itself, so the thread.js code does not have to do anything

Reply:
The issue is that the confirmation dialog displays a `Message` component (hence the custom dialog). This would lead to cyclic dependencies: `message -> confirmationDialog --> message`.  If there a standard solution for this scenario ?

Reply:
Hmm, indeed. if the custom dialog has to display the message, then I think confirmation dialog itself will not work.  Maybe we can ask ngr to see if it is really necessary to display the message, and not just the body.

But if we need, then yes, a custom dialog is probably required. Now, for the circular dependency, it is indeed annoying, but having code in thread that should be in message is even more annoying. A workaround is to give the `Message` component itself in props to the custom dialog (and then, the custom dialog can do a `<t t-component="props.Message" />` instead of `<Message />`)

Reply:
We could solve this issue by having a single point of entry for imports.
And this point of entry makes all necessary imports/exports.

This solution would solve any cyclic dependency (even `A` -> `B` -> `C` -> `A` patterns), allow keeping `import/export`, easy-to-maintain split of files in any folder we want, and also no need for workaround like rely on code evaluation like here with prop passing + `t-component` directive.

Reference: https://medium.com/visual-development/how-to-fix-nasty-circular-dependency-issues-once-and-for-all-in-javascript-typescript-a04c987cf0de